tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ActivityTemplateConverterTests.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Templates;
+using Microsoft.Bot.Schema;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Tests
+{
+    [TestClass]
+    public class ActivityTemplateConverterTests
+    {
+        [TestMethod]
+        [DataRow("{'activity': 'blah'}")]
+        public void ReadsActivityTemplateJson(string jsonText)
+        {
+            // TODO: check with tom why this test case doesn't work:
+            // [DataRow("{'activity': { '$kind': 'Microsoft.ActivityTemplate', 'template': 'blah'}")]
+            var jsonReader = CreateJsonReader(jsonText);
+            var sut = new ActivityTemplateConverter();
+            var result = sut.ReadJson(jsonReader, null, null, null);
+            Assert.IsInstanceOfType(result, typeof(ActivityTemplate));
+            var template = (ActivityTemplate)result;
+            Assert.AreEqual("blah", template.Template);
+        }
+
+        [TestMethod]
+        public void ReadsStaticActivityTemplateActivityTemplateJson()
+        {
+            var jsonText = @"{'activity': {
+            '$kind': 'Microsoft.StaticActivityTemplate',
+            'activity': {
+                'type': 'event',
+                'name': 'BookFlight'
+            }
+            }}";
```

Comment:
This is wrong.  What we are modeling is that

prompt => "Hello world"  
prompt => {  ... ACTIVITY .... }
prompt => {  "$kind":"Sometype which implements ITemplate<Activity>" ....

Reply:
See tests\Microsoft.Bot.Builder.TestBot.Json\Samples\80 - AdaptiveSkillDialog\AdaptiveSkillDialog.main.dialog this is what the test is based on.

Reply:
Updated AdaptiveSkillDialog.main.dialog to use the simpler notation and manually tested it verify it works after merging latest from master with the new serialization and schema updates.